js/app.js

```diff
@@ -29,3 +26,40 @@ rawData.forEach(function(projObject) {
 projArr.forEach(function(project) {
   $('#projects').append(project.toHtml());
 });
+
+Project.loadAll = function(rawData) {
+  rawData.sort(function(a,b) {
+    return (new Date(b.publishedOn)) - (new Date(a.publishedOn));
+  });
+
+  rawData.forEach(function(ele) {
+    Project.all.push(new Project(ele));
+  })
+}
+
+Project.fetchAll = function() {
+  var serverETag;
+
+  $.ajax({
+    url: 'https://api.github.com/users/ragnaroksedge/repos?callback=?',
+    type: 'HEAD',
+    success: function(data, message, xhr) {
+      serverETag = xhr.getResponseHeader('ETag');
+    },
+    fail: function (err) {
+      console.error(err);
+    }
+  });
+
+  if (localStorage.rawData && localStorage.ETag === serverETag) {
+    Project.loadAll(JSON.parse(localStorage.rawData));
+    projectView.initIndexPage();
+  } else {
+    $.getJSON('/../data/hackerIpsum.json', function(data) {
```

Comment:
You're still calling a local json file that does not exist in your project. And you're still using `project.js`. Try to change that to a local `project.json` file so that you can do `getJson` and pass in your `project.json` location.